README-ocp4-cs.md

```diff
@@ -0,0 +1,388 @@
+4](README-ocp4.md)
+
+A note on namespaces: The KubeFed operator itself supplies an option to either
+namespace the operator or install cluster-wide. This is a different concept
+from KubeFed itself being either namespace or cluster scoped. If you choose
+`All namespaces on the cluster` during the **operator** installation, it will
+install in the `openshift-operators` namespace, and you will need to supply the
+argument `--kubefed-namespace=openshift-operators` to all invocations of the
+`kubefedctl` command in this document. These instructions will instead guide
+you through installing the operator to the `kube-federation-system` namespace,
+which is the default namespace for the `kubefedctl` command to store its objects.
+The namespace must be created on the command line because the web UI considers
+namespaces starting with `kube` to be protected.
+
+0. Create the `kube-federation-system` namespace.
+   1. `oc create ns kube-federation-system`
+1. Login into `cluster1` web console as `kubeadmin` user.
+   1. Login details are reported by the installer in the file `auth/kubeadmin-password`.
+2. Install KubeFed from `Operator Hub`.
+   1. On the left panel click `Catalog -> Operator Hub`.
+   2. Select `KubeFed` from operator list.
+   3. If a warning about use of Community Operators is shown click `Continue`.
+   4. Click `Install`.
+   5. Under `Installation Method`, Select A `specific namespace on the cluster`
+   6. Select `kube-federation-system` as the namespace.
+   7. Click `Subscribe`.
+3. Check the Operator Subscription status.
+   1. On the left panel click `Catalog -> Operator Management`.
+   2. Click `Operator Subscriptions` tab.
+   3. Ensure the `Status` is "Up to date" for the `kubefed-operator` subscription.
+4. Create a KubeFed resource to instantiate the KubeFed controller.
+   1. On the left panel click `Catalog -> Installed Operators`.
+   2. Click `Kubefed Operator`.
+   3. Under `Provided APIs`, find `KubeFed`, and click `Create New`.
+   4. Change the `spec: scope:` from `Namespaced` to `Cluster`.
+   5. Click `Create`
+
+If everything works as expected, there should be a kubefed-controller-manager
+Deployment with two pods running in `kube-federation-system`.
+
+~~~sh
+oc --context=cluster1 -n kube-federation-system get pods
+
+NAME                                          READY   STATUS    RESTARTS   AGE
+kubefed-controller-manager-67d5d5cc99-j9zh9   1/1     Running   0          1m
+kubefed-controller-manager-67d5d5cc99-s59gx   1/1     Running   0          1m
+kubefed-operator-694c8f9cdc-9pc5z             1/1     Running   0          2m
+
+~~~
+
+Now we are going to enable some of the federated types needed for our demo
+application. You can watch as FederatedTypeConfig resources are created for
+each type by visiting the `All Instances` tab of the `Kubefed Operator` under
+`Catalog -> Installed Operators`.
+
+~~~sh
+for type in namespaces secrets serviceaccounts services configmaps deployments.apps
```

Comment:
In order to demonstrate the cluster scoped federation, we should enable a cluster-wide type like ClusterRole.

What do you think about enabling a ClusterRole and then creating a FederatedClusterRole within the demo-app?

Reply:
Yes! I forgot I wanted to add this before merging. I'll put the WIP label back on.